stockflux-securities-master/src/components/form-view/form/Form.js

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import { AlertType } from '../../alerts/AlertType';
+import ToggleSwitch from '../form-controls/toggle-switch/ToggleSwitch';
+import TextField from '../form-controls/text-field/TextField';
+import * as action from '../../../actions/Security';
+import * as service from '../../../services/SecuritiesService';
+import securityReducer, { initialState } from '../../../reducers/Security';
+import './Form.css';
+
+const Form = ({ securityId, setAlerts, setRedirect }) => {
+  const [state, dispatch] = useReducer(securityReducer, initialState);
+
+  useEffect(() => {
+    if (securityId) {
+      service
+        .getSecurity(securityId)
+        .then(security => {
+          setSecurity(security);
+        })
+        .catch(() => {});
+    }
+  }, [securityId]);
+
+  const setSecurity = security => {
+    dispatch(action.setName(security.name));
+    dispatch(action.setExchange(security.exchange));
+    dispatch(action.setSymbol(security.symbol));
+    // TODO: Change `!security.enabled` to `security.disabled` once BE is updated
+    dispatch(action.setDisabled(!security.enabled));
+  };
+
+  const processResponse = async (response, successStatusCode) => {
+    if (response.status === successStatusCode) {
+      setRedirect(true);
+    } else {
+      await populateAlerts(response);
+    }
+  };
+
+  const submitEdit = securityId => {
+    service
+      .updateSecurity(securityId, getSecurityDTO())
+      .then(async response => await processResponse(response, 200));
+  };
+
+  const submitNew = () => {
+    service
+      .postSecurity(getSecurityDTO())
```

Comment:
The presence of this in this component further reinforces that this should be component state and not in the store. you can post this with useEffect here if it was in component state.